scout/server/blueprints/diagnoses/static/diagnoses.js

```diff
@@ -0,0 +1,153 @@
+async function loadDiseases() {
+		//Remove load button
+    let button=document.getElementById("load-button")
+
+    button.remove()
+
+    //Add spinner to show loading is in progress
+    $('#load-container').html(
+        `<div id="spinner-container" class="d-flex align-items-center">
+            <div class="spinner-border text-primary m-3" role="status">
+                <span class="visually-hidden"></span>
+            </div>
+           Loading all diagnoses, this might take some time...
+        </div>`);
+
+    const baseUrl = window.location.origin
+
+    try {
+        //Fetch data and generate table with results
+        const response = await fetch(`${baseUrl}/api/v1/diagnoses`)
```

Comment:
I confess I had in mind something that would not involve waiting times for this page. I would have loaded no terms initially (as you did) and then, instead of loading them all and filtering using the functionality already present in the page, I would have used the same disease query form that is present in the case page when you assign case diagnoses:

<img width="331" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/882db559-561c-4047-a852-4f14da9876f1">

Or a query endpoint similar to that that, returns the complete info for the diseases

Your way works even if it is a bit slow (it's still faster than the mess we have on the page now), so I think it is good enough for now

Reply:
The page is aimed at choosing a disease to associate to a case.

The table you have created with the list of diseases with links is perfect:

<img width="1035" alt="image" src="https://github.com/Clinical-Genomics/scout/assets/28093618/1172e7cc-73f0-4d59-aea9-3887a76e200a">

The users are not interested in browsing the available terms, but they can't run queries if they don't load all terms first, so choosing a disease for a case takes time. 

Would be simpler to have a form where the user searches for a string and gets the same results as now, but without having to load all terms first